packages/shared/src/components/auth/ForgotPasswordForm.tsx

```diff
@@ -1,32 +1,85 @@
 import classNames from 'classnames';
-import React, { ReactElement, useState } from 'react';
+import React, { FormEvent, ReactElement, useState } from 'react';
+import { useMutation, useQuery } from 'react-query';
+import {
+  AccountRecoveryParameters,
+  getErrorMessage,
+  getNodeByKey,
+  getNodeValue,
+  ValidateRecoveryParams,
+} from '../../lib/auth';
+import { formToJson } from '../../lib/form';
+import { disabledRefetch } from '../../lib/func';
 import { Button } from '../buttons/Button';
 import { TextField } from '../fields/TextField';
 import MailIcon from '../icons/Mail';
 import VIcon from '../icons/V';
 import { CloseModalFunc } from '../modals/common';
 import AuthModalHeader from './AuthModalHeader';
-import { AuthModalText } from './common';
+import { AuthForm, AuthModalText } from './common';
+import TokenInput from './TokenField';
 import useTimer from '../../hooks/useTimer';
+import {
+  AuthFlow,
+  initializeKratosFlow,
+  submitKratosFlow,
+} from '../../lib/kratos';
 
 interface ForgotPasswordFormProps {
-  email?: string;
+  initialEmail?: string;
   onBack?: CloseModalFunc;
   onClose?: CloseModalFunc;
 }
 
 function ForgotPasswordForm({
-  email,
+  initialEmail,
   onBack,
   onClose,
 }: ForgotPasswordFormProps): ReactElement {
+  const [hint, setHint] = useState('');
+  const [sentCount, setSentCount] = useState(0);
   const [emailSent, setEmailSent] = useState(false);
   const { timer, setTimer, runTimer } = useTimer(() => setEmailSent(false), 0);
+  const { data: recovery } = useQuery(
+    ['recovery', sentCount],
+    () => initializeKratosFlow(AuthFlow.Recovery),
+    {
+      ...disabledRefetch,
+    },
+  );
+
+  const { mutateAsync: sendEmail, isLoading } = useMutation(
+    (params: ValidateRecoveryParams) => submitKratosFlow(params),
+    {
+      onSuccess: ({ error }) => {
+        if (error) {
+          const requestError = getErrorMessage(error.ui.messages);
+          const emailError = getNodeByKey('email', error.ui.nodes);
+          const formError = getErrorMessage(emailError?.messages);
```

Comment:
Should we perhaps consider abstracting this in a common function?
Not sure if it's a once-off at this stage.

non blocking.

Reply:
I think I haven't had the chance that I had to use this one. Though this one is specific for this email form.